### ISSUE TITLE
[BUG] Bizzarre Satify Number behavior

### ISSUE THREAD

**Issue by Bram28** [Bram-Hub/LEGUP#555]
### Description

OK, *really* bizarre one ... I have a case rule that checks out when it should clearly not

### Steps to Reproduce

First I do some initial moves. So far so good:

![image](https://github.com/Bram-Hub/LEGUP/assets/2414519/927f02de-2dda-474d-a8cf-f0019278639e)

Now I put a lightbulb to the right of the 2 ... and select the Case Rule 'Satify Number':

![image](https://github.com/Bram-Hub/LEGUP/assets/2414519/fe6d08a4-5a66-4a29-ad94-18c616376b1f)

Oops!  This should not check out. And strangely, it does not check out when the very top right square is not lit up. For example:

![image](https://github.com/Bram-Hub/LEGUP/assets/2414519/01bfaec5-ff97-460e-ae5d-d5f95b3c4e06)

I really have no idea what could lead to this behavior ... someone will have to trace this

### Expected Behavior

As explained above, the rule should not check out

**Comment by charlestian23** [comment]
Ok, after a multitude of testing this week, I have come to the conclusion that this is a general bug with case rules. I believe once a transition is present like this:
![image](https://github.com/Bram-Hub/LEGUP/assets/46334090/ec31a6ed-88bd-4bae-8a7a-38a5508ae08c)
even if there are no actual changes present on the board, the case rule checker starts to act strangely. This is because many of the case rules rely on `parent.getChildren()` for error checking. By making a modification to the board, we set the number of children to 1. However, in the case of Satisfy Number, you are allowed to have 1 child, and so this results in the rule missing the error check.

I have also verified this is an engine-wide bug with testing on Nurikabe:
1. Open any LEGUP Nurikabe puzzle
2. Make any modification to the board
3. Click on the cells you modified and change them back to unknown. At this point, there should be no difference between the starting board and the board currently on your screen, but there should exist a transition on the tree in the TreeView.
4. Attempt to use the Black or White case rule, which should be allowed to be used at this point.
5. Note that LEGUP creates a new node with a red transition.

A possible solution I will explore is deleting this arrow on any call of a case rule. I'm not sure how this is possible right now and will need to do more testing to figure out how to implement this.
